mtgjson4/outputter.py

```diff
@@ -165,6 +166,27 @@ def create_all_cards(files_to_ignore: List[str]) -> Dict[str, Any]:
     return all_cards_data
 
 
+def create_all_cards_subsets(
+    cards: Dict[str, Any], target_formats: Set
+) -> Dict[str, Any]:
+    """
+    For each format in target_formats, a dictionary is created containing only cards legal in that format. Legal
+    includes cards on the restricted list, but not those which are banned.
+
+    :param cards: The card data to be filtered (should be AllCards)
+    :param target_formats: The set of formats to create subsets for.
+
+    :return: Dictionary of dictionaries, mapping format name -> all_cards subset.
+    """
+    subsets: Dict[str, Dict[str, Any]] = {fmt: {} for fmt in target_formats}
+    for name, data in cards.items():
+        for fmt in target_formats:
+            if data["legalities"].get(fmt) in ("Legal", "Restricted"):
```

Comment:
Good change to do `if var in set(...)` it seems to be much faster than `if var in list(...)`. I should go back and start profiling this project...